File: 2016_11_1/insertion_sort/insertion_sort.js
#!/usr/bin/node

function insertion(a) {
	for (let i = 1; i < a.length; i++) {
		let t = a[i];
		let k = i;
		while (k > 0 && t < a[k-1]) {
			a[k] = a[k-1];
			k--;
		}
		a[k] = t;
	}
}

let size = parseInt(process.argv[2]);
let test = [];
for (let i = 0; i < size; i++)
	test.push(i);
for (let i = 0; i < size; i++) {
	let r = Math.floor(Math.random() * (size - i)) + i;
	[test[i],test[r]] = [test[r],test[i]];
}
console.log(test);
insertion(test);
console.log(test);

File: 2016_11_04/insertionsort/insertionsort.py
#!/usr/bin/python3

import random, sys, time

def insertion(a):
	for i in range(1, len(a)):
		m = a[i]
		k = i
		for j in range(i, 0, -1):
			if m < a[j-1]:
				a[j] = a[j-1]
				k -= 1
		a[k] = m

size = int(sys.argv[1])
test = list(range(size))
random.shuffle(test)
print("===============================================================================")
if size < 32: print(test)
begin = time.time()
insertion(test)
end = time.time()
if size < 32: print(test)
print("Time:", int(round((end - begin))), "s")
print("Time:", int(round((end - begin) * 1000)), "ms")
print("Time:", int(round((end - begin) * 1000000)), "μs")
print("Time:", int(round((end - begin) * 1000000000)), "ns")
print("===============================================================================")

